rskj-core/src/main/java/co/rsk/net/BlockSyncService.java

```diff
@@ -105,6 +103,17 @@ public BlockProcessResult processBlock(@Nonnull Block block, Peer sender, boolea
                 logger.trace("Missing hashes for block in process {} {}", blockNumber, block.getPrintableHash());
                 requestMissingHashes(sender, unknownHashes);
             }
+            return false;
+        }
+
+        return true;
+    }
+
+    public BlockProcessResult processBlock(@Nonnull Block block, Peer sender, boolean ignoreMissingHashes) {
+        final Instant start = Instant.now();
+
+        boolean looksGood = preprocessBlock(block, sender, ignoreMissingHashes);
```

Comment:
"preprocess" is executed twice? first time when you put in the queue and this is the second time when you actually process it?

Reply:
Yes

Reply:
We could leave this until the next step. It's too soon to take care of this case.